packages/langium/src/workspace/documents.ts

```diff
@@ -304,18 +311,39 @@ export class DefaultLangiumDocuments implements LangiumDocuments {
         let langiumDoc = this.documentMap.get(uriString);
         if (langiumDoc) {
             // The document is already present in our map
-            return langiumDoc;
+            if (langiumDoc.state < DocumentState.Parsed) {
+                this.reloadDocumentContent(langiumDoc);
```

Comment:
This change adds implicit behavior to this method, which I would prefer to make explicit: Add a protected method to DocumentBuilder that is called on `Changed` documents, invokes the parser and assigns a new `parseResult`.